src/explorer/trialApp/TrialAppClient.ts

```diff
@@ -51,17 +53,50 @@ export class TrialAppClient implements IFilesClient {
         return `https://${this.metadata.scmHostName}`;
     }
 
-    public get defaultHostName(): string {
-        return this.metadata.hostName;
-    }
-
     public get defaultHostUrl(): string {
-        return `https://${this.metadata.hostName}`;
+        return `https://${this.fullName}`;
     }
 
     public async getKuduClient(): Promise<KuduClient> {
         const kuduClient: KuduClient = new KuduClient(this._credentials, this.kuduUrl);
         addExtensionUserAgent(kuduClient);
         return kuduClient;
     }
+
+    public async listApplicationSettings(): Promise<StringDictionary> {
+        const kuduClient: KuduClient = await this.getKuduClient();
+        const settings: StringDictionary = {};
+        settings.properties = <{ [name: string]: string }>await kuduClient.settings.getAll();
+        return settings;
+    }
+
+    public async updateApplicationSettings(appSettings: StringDictionary): Promise<StringDictionary> {
```

Comment:
This definitely all seems kind of weird, but after your offline discussion, I get why you have to do it.

Reply:
Yeah this is not great. For example, there's a possibility of getting into a weird half-updated state if one of the deletes fails before you even try the update.

I think the only reason I'm okay with this is because it's a trial app that'll get thrown away after an hour anyways (aka it's not a big deal if settings get messed up by us). That being said, if you're doing "weird" things in the code, you should generally add comments (to the TypeScript file itself, not on GitHub) describing why you're doing it.

Reply:
Ok, I can add a comment explaining the weird code. I also think we are going to filter out many of the app settings in the future because lots of the trial app settings are not things we want to show to users.

Reply:
I actually already have a comment explaining the code [here](https://github.com/microsoft/vscode-azureappservice/pull/1588/files/5eff536c5352854732d17d227913860733e88618#diff-6b9c6c17562552f98bb7ff2484af2216R76).

Reply:
That comment is _what_ you're doing, not _why_ you're doing it. A few questions the comment should answer:

- Why are you making your own requests instead of using KuduClient or SiteClient?
- Why isn't rename automatically handled for you like normal apps?

Your mentor needed an offline discussion to understand why you're doing this, so obviously I'm not the only one that wants more context :P

Reply:
Also it's fine to say something along the lines of "I don't know the root cause, but I tested X and Y happened" in your comment. This is all meant to help someone reading your code down the line. Maybe they read your comment, think of something you didn't try, and fix it. Maybe they try what you tried, get a different result, and realize the platform must've changed

Worst case is they look at your code, think it looks ugly, refactor to follow the same practice as normal apps, don't test a trial app (because it's annoying to set up and why would it be different? lol), and now we have broken app settings for trial apps